core/pkg/store/store.go

```diff
@@ -50,49 +52,79 @@ func (f *State) hasPriority(stored string, new string) bool {
 	return true
 }
 
-func NewFlags() *State {
-	return &State{
-		Flags:             map[string]model.Flag{},
+func NewStore(logger *logger.Logger) (*Store, error) {
+
+	schema := &memdb.DBSchema{
+		Tables: map[string]*memdb.TableSchema{
+			"flags": {
+				Name: "flags",
+				Indexes: map[string]*memdb.IndexSchema{
+
+					"id": {
+						Name:    "id",
+						Unique:  true,
+						Indexer: &memdb.StringFieldIndex{Field: "Key", Lowercase: false},
+					},
```

Comment:
Currently we just have a simple index on `Key`, to mimic the previous map behavior, but `go-memdb` supports all kinds of index types very useful to us, including compound indexes, numeric indexes, conditional indexes, etc. We can use these to great effect.